lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,256 @@
+    required: False
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Configure DNS for an host system
+  vmware_host_dns:
+    hostname: vcenter01.fqdn.local
+    username: administrator@vsphere.local
+    password: SuperSecretPassword
+    esxi_hostname: esx01.fqdn.local
+    host_name: esx01
+    domain: fqdn.local
+    dns_servers:
+      - 192.168.1.10
+      - 192.168.1.11
+    search_domains:
+      - subdomain.fqdn.local
+      - fqdn.local
+  delegate_to: localhost
+
+- name: Configure DNS for all host systems from cluster
+  vmware_host_dns:
+    hostname: vcenter01.fqdn.local
+    username: administrator@vsphere.local
+    password: SuperSecretPassword
+    cluster_name: cluster01
+    domain: fqdn.local
+    dns_servers:
+      - 192.168.1.10
+      - 192.168.1.11
+    search_domains:
+      - subdomain.fqdn.local
+      - fqdn.local
+  delegate_to: localhost
+'''
+
+# TODO: update output
+RETURN = r'''
+result:
+  description: metadata about host system's DNS configuration
+  returned: always
+  type: dict
+  sample: {
+    "esxi01": {
+    "msg": "DNS is already enabled and active for host 'esxi01'",
+    },
+  }
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostDNS(PyVmomi):
+    """Class to manage DNS configuration of an ESXi host system"""
+    def __init__(self, module):
+        super(VmwareHostDNS, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """Function to manage DNS configuration of an ESXi host system"""
+        results = dict(changed=False, result=dict())
+        cluster_name = self.params.get('cluster_name', None)
+        if not cluster_name:
+            host_name = self.params.get('host_name')
+        domain = self.params.get('domain')
+        dns_servers = self.params.get('dns_servers')
+        search_domains = self.params.get('search_domains', None)
+        # NOTE: The below block is also deprecated starting from Ansible v2.11
+        if self.params.get('domainname') is not None:
+            domain = self.params.get('domain')
```

Comment:
add an alias to `domain` as `domainname`

Reply:
This is also to deprecate parameter `domainname`.

Reply:
These are ghost options from `vmware_dns_config`.